src/test/functions/functionsLog.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { expect } from "chai";
+import * as sinon from "sinon";
+import * as functionsLog from "../../functions/functionslog";
+import { logger } from "../../logger";
+
+describe("functionsLog", () => {
+  describe("getApiFilter", () => {
+    it("should return base api filter for v1 functions", () => {
+      expect(functionsLog.getApiFilter(false, undefined)).to.eq('resource.type="cloud_function"');
+    });
+
+    it("should return base api filter for v1&v2 functions", () => {
+      expect(functionsLog.getApiFilter(true, undefined)).to.eq(
+        'resource.type="cloud_function" OR (resource.type="cloud_run_revision" AND labels."goog-managed-by"="cloudfunctions")'
+      );
+    });
+
+    it("should return list api filter for v1 functions", () => {
+      expect(functionsLog.getApiFilter(false, "fn1,fn2")).to.eq(
+        'resource.type="cloud_function"\n(resource.labels.function_name="fn1" OR resource.labels.function_name="fn2")'
+      );
+    });
+
+    it("should return list api filter for v1&v2 functions", () => {
+      expect(functionsLog.getApiFilter(true, "fn1,fn2")).to.eq(
+        'resource.type="cloud_function" OR (resource.type="cloud_run_revision" AND labels."goog-managed-by"="cloudfunctions")\n(resource.labels.function_name="fn1" OR resource.labels.service_name="fn1" OR resource.labels.function_name="fn2" OR resource.labels.service_name="fn2")'
```

Comment:
nit* Would love to see us break this into multiline string 

e.g.

```
'resource.type="cloud_function" OR +
  '(resource.type="..)
...
```

Reply:
broke out into a multiline string